lib/parsers.js

```diff
@@ -88,16 +88,21 @@ exports.parseError = function (err, agent, cb) {
   })
 }
 
-exports.getContextFromRequest = function (req, captureBody) {
+exports.getContextFromRequest = function (req, conf, type) {
+  var captureBody = conf.captureBody === type || conf.captureBody === 'all'
+
   var context = {
     http_version: req.httpVersion,
     method: req.method,
     url: getUrlFromRequest(req),
     socket: {
       remote_address: req.socket.remoteAddress,
       encrypted: !!req.socket.encrypted
-    },
-    headers: Object.assign({}, req.headers)
```

Comment:
If we set `headers: undefined` here, we'll reduce the number of hidden classes by one right?

Reply:
fixed.